lib/KCheckbox/index.vue

```diff
@@ -178,17 +255,23 @@
     methods: {
       toggleCheck(event) {
         if (!this.disabled) {
+          this.lastUserEvent = event || null;
           this.$refs.kCheckboxInput.focus();
-          /**
-           * Emits change event
-           */
-          this.$emit('change', !this.checked, event);
+          this.isChecked = !this.isChecked;
+          this.lastUserEvent = null;
```

Comment:
At first read, it is not apparent what the `lastUserEvent` is for here, or why it is set to null again. We should add a couple of comments to explain this better.

However, to play safe, I would suggest setting the `lastUserEvent` to null inside a [nextTick](https://vuejs.org/api/general#nexttick) call, just to make sure that this event is being set to null just after the change event has already been emitted, and not before. Another alternative to actually make sure that this is set to null just after the change event has been emitted is to set it to null within the `updateInputValue` method. Since it would mean "okay, I already used it, we don't need this anymore", and this will ensure that this value is removed just after the change event has been emitted.